integration/spark/src/main/spark3.1/org/apache/spark/sql/CarbonToSparkAdapter.scala

```diff
@@ -180,9 +180,16 @@ object CarbonToSparkAdapter extends SparkVersionAdapter {
       }
   }
 
-  def getDataFilter(partitionSet: AttributeSet, filter: Seq[Expression]): Seq[Expression] = {
+  def getDataFilter(partitionSet: AttributeSet, filter: Seq[Expression],
+      partitionFilter: Seq[Expression]): Seq[Expression] = {
     filter.filter {
-      case _: DynamicPruningSubquery => false
+      case dp: DynamicPruningSubquery =>
+        // if filter does not exists in partition filter, then push down the filter to spark
+        if (partitionFilter.exists(_.semanticEquals(dp))) {
```

Comment:
no need of if else, directly return  !partitionFilter.exists(_.semanticEquals(dp))